src/game_actor.cpp

```diff
@@ -114,6 +114,12 @@ bool Game_Actor::IsSkillUsable(int skill_id) const {
 	if (skill_id <= 0 || skill_id > (int)Data::skills.size()) {
 		return false;
 	}
+	for (auto inflectedState : this->GetInflictedStates()) {
+		if ((Data::states[inflectedState - 1].restrict_magic && Data::skills[skill_id].magical_rate > Data::states[inflectedState - 1].restrict_magic_level) || ((Data::states[inflectedState - 1].restrict_skill && Data::skills[skill_id].hit > Data::states[inflectedState - 1].restrict_skill_level))) {
+			return false;
+		}	
+	}
+	
```

Comment:
inflicted, not inflected :)
Because the data type is simple to write you can also use "int" instead of "auto".

skill_id is off-by-one, needs a -1